src/test/cs/Infinispan/HotRod/ContinuousQueryTest.cs

```diff
@@ -0,0 +1,383 @@
+        }
+        [TestFixtureTearDown]
+        public void AfterClass()
+        {
+            remoteManager.Stop();
+        }
+        [Test]
+        public void EntityBasicContQueryTest()
+        {
+            int joined = 0, updated = 0, leaved = 0;
+            try
+            {
+                IRemoteCache<int, User> userCache = remoteManager.GetCache<int, User>(NAMED_CACHE);
+                Semaphore s = new Semaphore(0, 1);
+                QueryRequest qr = new QueryRequest();
+                // JpqlString will be deprecated please use QueryString
+                // qr.JpqlString = "from sample_bank_account.User";
+                qr.QueryString = "from sample_bank_account.User";
+
+                Func<object, User> vFromByteBuffer = (b) => new Google.Protobuf.MessageParser<User>(() => new User()).ParseFrom((byte[])b);
+                Func<object, int> kFromByteBuffer = (b) => new Google.Protobuf.MessageParser<base_types>(() => new base_types()).ParseFrom((byte[])b).I32;
+
+                Event.ContinuousQueryListener<int, User> cql = new Event.ContinuousQueryListener<int, User>(qr.QueryString, kFromByteBuffer, vFromByteBuffer);
+                cql.JoiningCallback = (int k, User v) => { joined++; };
+                cql.LeavingCallback = (int k, User v) => { leaved++;  };
+                cql.UpdatedCallback = (int k, User v) => { updated++; s.Release(); };
+                userCache.AddContinuousQueryListener(cql);
+
+                User u = CreateUser1(userCache);
+                userCache.Put(1, u);
+                u.Name = "Jerry";
+                u.Surname = "Mouse";
+                userCache.Put(1, u);
+                userCache.Remove(1);
+                s.WaitOne(10000);
+                userCache.RemoveContinuousQueryListener(cql);
+                userCache.Clear();
+            }
+            catch (System.Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
+            Assert.AreEqual(1, joined);
+            Assert.AreEqual(1, updated);
+            Assert.AreEqual(1, leaved);
+        }
+        [Test]
+        public void ProjectionBasicContQueryTest()
+        {
+            int joined = 0, updated = 0, leaved = 0;
+            Tuple<int,string> uT = null, lT = null;
+            try
+            {
+                IRemoteCache<int, User> userCache = remoteManager.GetCache<int, User>(NAMED_CACHE);
+                userCache.Clear();
+                Semaphore s = new Semaphore(0, 1);
+                QueryRequest qr = new QueryRequest();
+                qr.QueryString = "select id, name from sample_bank_account.User";
+                Google.Protobuf.MessageParser < base_types > mp = new Google.Protobuf.MessageParser<base_types>(() => new base_types());
+                Google.Protobuf.MessageParser<base_types> mp1 = new Google.Protobuf.MessageParser<base_types>(() => new base_types());
```

Comment:
The mp and mp1 types are completely same. Could we use just one variable of that type?